[Shaun] IP

Hoodini231 opened the pull request:
### Meow Chatbot

**Are** you?
- Unhappy with your life?
- Feeling really lonely?
- Need a task list?
- Really like cats?

WELL Meow is here to solve all of that !

> "Without Meow chatbot and taskmanager I wouldn't Have gotten 5.0 GPA" ~Gong Yi CS1101S TA

> "I would have been so lonely without Meow, thank goodness for Meow" ~Vice Dean A/P Gary

All **you** need to do is:

1. Download the [file](https://github.com/nus-cs2103-AY2425S1/ip)
2. Run the file
3. Add your task using CLI
4. Let it meow meow ur tasks for u! 😉

Features:

- [ ] Meowceptions to tell u what u did wrong
- [ ] Manage Tasks
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

````java
public static final void main(String[] args) throws Meowception, IOException {
    new Meow().run();
}
`

Hoodini231 committed:
Renamed to Meow.java, added new logo and greeting.
Level-0 completed.

Hoodini231 committed:
Add new feature of echoing the users input.
Infinite user loop until user inputs 'bye'.
Level 1 completed.

Hoodini231 committed:
Add functional list. Supported ability for user
to add tasks via String input. And To show tasks
via the list command.
Level-2 Completed.

Hoodini231 committed:
Add Task class to help manage tasks and data.

Hoodini231 committed:
Add support for functional mark / unmark list.
Added some error handling, to validate inputs.
Level 3 done.

Hoodini231 committed:
Add Deadline, Event, Todo classes to enforce
speical and unique types of tasks. Added error
handling, and input validation to ensure
Deadlines, event and todo tasks are properly formatted
and inputted. Thus, no blanks, or missing fields.

Hoodini231 committed:
Add support for automated ui testing via runtest,
Simple test which checks for adding a todo task,
marking it and whether it throws an error.

Hoodini231 committed:
Added support for new exception handling called
Meowceptions. Support for unmark and mark completed.

Hoodini231 committed:
Add error handling support to deadline tasks.

Hoodini231 committed:
Add error handling support for Event task

Hoodini231 committed:
Add support for Meowception 001 error

Hoodini231 committed:
Add delete feature to remove a task. Added support
for error handling for feature also

Hoodini231 committed:
Add enums and switch cases to handle user commands
Enums functinal.

Hoodini231 committed:
Add support to save tasks to the local disk

Hoodini231 committed:
Add support for DateTime objects

Hoodini231 committed:
Fix the bug with Meowception

Hoodini231 committed:
Support for OOP, add Parser, TaskList, Ui

Hoodini231 committed:
Support for OOP done, functional and reworked

Hoodini231 committed:
Fix readability and code style

Hoodini231 committed:
Add packages into the project

Hoodini231 committed:
Add gradle support

Hoodini231 committed:
Support for Gradle fixed and Junit added

Hoodini231 committed:
Updated files in test for Junit

Hoodini231 committed:
Add support for Jar file

Hoodini231 committed:
Add more JavaDocs comments

Hoodini231 committed:
Add more javadoc comments to code base

Hoodini231 committed:
Fix some coding standard issues

Hoodini231 committed:
Merge branch 'branch-A-CodingStandard'

Hoodini231 committed:
Add Find function and reformatted code

Hoodini231 committed:
Merge branch 'branch-Level-9'

Hoodini231 committed:
Add checkstyle configuration files

harithh07 commented on src/main/java/com/meow/Meowception.java:
```diff
@@ -0,0 +1,35 @@
+public class Meowception extends Exception {
+    String code;
+    
+    // public consturctor for Meowception
```
Is this a typo? Should there be more explanation for the constructor to be clearer?

harithh07 commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+     * @param String inputType
+     * @return String
+     */
+    public String commandValidation(String inputType) throws Meowception {
```
Should be this named as a verb instead? Eg. checkValidCommand or validateCommand

choaticman replied:
Perhaps a verb phrase name such as validateCommand would be clearer.

Should the parameter be named inputCommand instead? This could be clearer in inferring that the input is.

harithh07 commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+     * @param String inputType
+     * @return String
+     */
+    private String outputTask(Commands cmd,String inputType) throws Meowception {
```
Is this missing a space between params?

choaticman replied:
Should there be a space between the parameters? 

```suggestion
    private String outputTask(Commands cmd, String inputType) throws Meowception {
```

harithh07 commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+     * @return String
+     */
+    private String outputTask(Commands cmd,String inputType) throws Meowception {
+        //System.out.println(inputType.substring(5).trim());
```
Should this still be here?

choaticman replied:
Should this commented out code still be here?

harithh07 commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+                    Meowception err = new Meowception("100");
+                    throw err;
+                }
+                // errorMsg("Meow meow you need to enter a task number to mark !!!");
```
Should this be here as well?

harithh07 commented on src/main/java/com/meow/Save.java:
```diff
@@ -0,0 +1,101 @@
+    /*
+     * Constructor for Save class
+     * @Params TaskList tasks
+     * @Return None
```
Should the @return comment still be here when its a void method?

harithh07 commented on src/test/java/EventTest.java:
```diff
@@ -0,0 +1,11 @@
+public class EventTest {
+    @Test
+    public void testTaskCreation() {
+        //Task task = new Event("Mewwing sesh", "");
```
Should this test still be here?

harithh07 commented on src/main/java/com/meow/com/tasks/Task.java:
```diff
@@ -0,0 +1,36 @@
+    public boolean isDone() {
+        return this.isDone;
+    }
+    public String getCompetionChar() {
```
Is this a typo?

harithh07 commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+     * Outputs the task based on the command
+     * @param Commands cmd
+     * @param String inputType
+     * @return String
```
Would it be clearer to explain the params and return types?

harithh07 reviewed (commented):
Quite a bit of commented out code throughout multiple files as well as some issues with the JavaDocs but otherwise, LGTM 👍

choaticman commented on src/main/java/com/meow/Meowception.java:
```diff
@@ -0,0 +1,35 @@
+package com.meow;
+public class Meowception extends Exception {
```
Good attempt in incorporating the name of the chatbot to classes' names. However, it may be confusing to some who may not be sure that Meowception is an Exception.

choaticman commented on src/main/java/com/meow/Meowception.java:
```diff
@@ -0,0 +1,35 @@
+            case "404":
+                return "Meowception ERROR 404: meow meow task not found meow";
+            default:
+                return "Unknown Meowception Error.";
```
Should there be indentation for the `case` clauses? Maybe you could configure your IDE to follow this style:

```suggestion
        switch (code) {
        case "001":
            return "Meowception ERROR 001: meow meow enter a valid command neow.";
        case "007":
            return "Meowception ERROR 007: meow meow file does not exist!";
        case "100":
            return "Meowception ERROR 100: meow meow you need to enter a task and it cant be blank silly goose meow";
        case "200":
            return "Meowception ERROR 200: meow meow you need to enter a deadline you stupido meow";
        case "300":
            return "Meowception ERROR 300: meow meow you need to enter a correct timefrime such as yyyy-mm-dd HHmm";
        case "404":
            return "Meowception ERROR 404: meow meow task not found meow";
        default:
            return "Unknown Meowception Error.";
```

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+                        throw new Meowception("100");
+                    }
+                } catch (Meowception err) {
+                    throw err;
```
Was the intended purpose to throw an error that is just caught? Perhaps the try-catch block would not be needed here.

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+    private String outputTask(Commands cmd,String inputType) throws Meowception {
+        //System.out.println(inputType.substring(5).trim());
+        switch (cmd) {
+            case MARK:
```
Should there be indentation for the `case` clauses? I noticed the same issue at several other classes too.

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+                        throw new Meowception("100");
+                    }
+                } catch (Meowception err) {
+                    throw err;     
```
Was the intended purpose to throw an error that was just caught?

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+                } 
+            case LIST:
+                return taskList.displayList();
+                // displayList();
```
Should this line still be here?

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+                try {
+                    Task todoTask = parseTodoTask(inputType.substring(5));
+                    return taskList.addTask(todoTask);
+                    //return taskList.addTodoTask(inputType.substring(5));
```
Should this line still be here? I noticed this in several other lines as well.

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+                break;
+            } 
+
+            throw new Meowception("001");
```
Should this be in a `finally` clause?

choaticman commented on src/main/java/com/meow/Meowception.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public String toString() {
+        switch (code) {
+            case "001":
```
Perhaps the use of an Enum could be clearer on what the error codes represent.

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    /*
+     * Parses the input command and returns a task
+     * @param String task
+     * @return String
+     */
```
Should this JavaDoc comment be here?

choaticman commented on src/main/java/com/meow/Parser.java:
```diff
@@ -0,0 +1,298 @@
+            return taskList.deleteTask(number);
+        }
+    }
+
+
+
+
+
+
+
+
```
Should these empty lines be here?

choaticman reviewed (commented):
Overall, I found your code easy to read. I noted a few minor coding standard violations
too. There are some commented out codes left inside as well.

Hoodini231 committed:
Add GUI to the chat bot
CSS added for styling sent and recieved messages

Hoodini231 committed:
Add javaDoc comments

Hoodini231 committed:
Fix code quality and style issues

Fixed issues relating to naming conventions, and removed many commented out code.

Hoodini231 committed:
Fix style issues

Hoodini231 committed:
Add assert support
Added asserts to validate behaviours that the user input should contain a task number and that number should be valid

Hoodini231 committed:
Fix code quality issues
Removed unnecessary imports, functions and commented out code further improved indentations and java doc comments

Hoodini231 committed:
Merge branch 'branch-A-CodeQuality'

Hoodini231 committed:
Fix issues with a merge resolve fix

Hoodini231 committed:
Add update extension feature
Added new command called update which allows uers to update details of different tasks.
Must use format update <task type> <task name> /<old or what they want to change> /new <new details>

Hoodini231 committed:
Add documentation and images
Wrote more detailed readMe and guide for users in using the meow bot. Furthermore, added images to the documentaiton.

Hoodini231 committed:
Add Ui.png

Hoodini231 committed:
Update README.md
add more feature descriptions within the readme page for users.

Hoodini231 committed:
Fix bugs and readme

Hoodini231 committed:
Fix bugs in Parser.java and update readme

Hoodini231 committed:
Fix bugs and issues
